test/system/internship_offer_search_desktop_test.rb

```diff
@@ -0,0 +1,180 @@
+    # reset search and submit
+    fill_in_keyword(keyword: '')
+    submit_form
+    assert_presence_of(internship_offer: searched_internship_offer)
+    assert_presence_of(internship_offer: not_searched_internship_offer)
+  end
+
+  test 'search by week works' do
+    travel_to(Date.new(2020,9,6)) do
+      searched_week = Week.selectable_from_now_until_end_of_school_year.first
+      not_searched_week = Week.selectable_from_now_until_end_of_school_year.last
+
+      searched_internship_offer = create(:weekly_internship_offer,
+                                         weeks: [searched_week])
+      not_searched_internship_offer = create(:weekly_internship_offer,
+                                             weeks: [not_searched_week])
+      visit internship_offers_path
+
+      select('3ème')
+      fill_in_week(week: searched_week, open_popover: true)
+      submit_form
+      assert_presence_of(internship_offer: searched_internship_offer)
+      assert_absence_of(internship_offer: not_searched_internship_offer)
+      # TODO: ensure weeks navigation and months navigation
+    end
+
+  end
+
+  test 'search by all criteria' do
+    searched_keyword = 'helloworld'
+    searched_week = Week.selectable_from_now_until_end_of_school_year.first
+    searched_location = Coordinates.paris
+    not_searched_keyword = 'bouhbouh'
+    not_searched_week = Week.selectable_from_now_until_end_of_school_year.last
+    not_searched_location = Coordinates.bordeaux
+    searched_opts = { title: searched_keyword,
+                      coordinates: searched_location,
+                      weeks: [searched_week]}
+    # build findable
+    findable_internship_offer = create(:weekly_internship_offer, searched_opts)
+
+    # build ignored
+    not_found_by_location = create(
+      :weekly_internship_offer,
+      searched_opts.merge(coordinates: Coordinates.bordeaux)
+    )
+    not_found_by_keyword = create(
+      :weekly_internship_offer,
+      searched_opts.merge(title: not_searched_keyword)
+    )
+    not_found_by_week = create(
+      :weekly_internship_offer,
+      searched_opts.merge(weeks: [not_searched_week])
+    )
+    not_found_by_school_track = create(
+      :bac_pro_internship_offer,
+      searched_opts.reject { |k,v| k == :weeks }
+    )
+
+    dictionnary_api_call_stub
+    SyncInternshipOfferKeywordsJob.perform_now
+    InternshipOfferKeyword.update_all(searchable: true)
+
+    visit internship_offers_path
+
+    fill_in_city_or_zipcode(with: 'Pari', expect: 'Paris')
+    fill_in_keyword(keyword: searched_keyword)
+    select('3ème')
+    fill_in_week(week: searched_week, open_popover: true)
+    submit_form
+
+    assert_presence_of(internship_offer: findable_internship_offer)
+    assert_absence_of(internship_offer: not_found_by_location)
+    assert_absence_of(internship_offer: not_found_by_keyword)
+    assert_absence_of(internship_offer: not_found_by_week)
+    assert_absence_of(internship_offer: not_found_by_school_track)
+  end
+end
```

Comment:
ajouter un test de persistence des champs saisis lors de la première recherche pour une deuxième recherche : a priori, on perd les champs saisis de ce que j'ai vu

Reply:
you say it ; you own it ;-)
PS @pierrax @fitchMitch j'ai qq tests en TODO ; je pense qu'il serait interessant que vous les fassiez au moins pr vous familiariser ac les tests (desktop/mobile)